frontend/src/views/CellGuide/components/CellGuideCard/components/MarkerGeneTables/hooks/computational_markers.ts

```diff
@@ -11,95 +15,129 @@ interface ComputationalMarkerGeneTableData {
   pc: string;
 }
 
-function _getSortedOrganisms(
-  genes: ComputationalMarkersQueryResponse
-): string[] {
-  const organisms = new Set<string>();
-  for (const markerGene of genes) {
-    if (markerGene.marker_score < FMG_GENE_STRENGTH_THRESHOLD) continue;
-    organisms.add(markerGene.groupby_dims.organism_ontology_term_label);
-  }
-  return Array.from(organisms).sort((a, b) => {
-    if (a === HOMO_SAPIENS) return -1;
-    if (b === HOMO_SAPIENS) return 1;
-    return a.localeCompare(b);
-  });
+enum selectionCriteriaErrorCode {
+  OrganismMismatch,
+  TissueMismatch,
+  LowMarkerScore,
+  Success,
 }
 
-function _getSortedOrgans(
-  genes: ComputationalMarkersQueryResponse,
-  selectedOrganismFilter: string
-): string[] {
-  const organs = new Set<string>();
-  for (const markerGene of genes) {
-    if (
-      markerGene.groupby_dims.organism_ontology_term_label !==
-      selectedOrganismFilter
-    )
-      continue;
-    if (markerGene.marker_score < FMG_GENE_STRENGTH_THRESHOLD) continue;
-    organs.add(
-      markerGene.groupby_dims.tissue_ontology_term_label ?? "All Tissues"
+// Apply selection criteria in the following order and short circut when a criteria fails:
+// 1. Organism
+// 2. Tissue
+// 3. Marker score
+function _applyOrderedSelectionCriteria({
+  markerGene,
+  allTissuesLabelToIdMap,
+  selectedOrganismLabel,
+  selectedOrganLabel,
+  selectedOrganId,
+}: {
+  markerGene: ComputationalMarkersQueryResponseEntry;
+  selectedOrganismLabel: string;
+  selectedOrganLabel: string;
+  selectedOrganId: string;
+  allTissuesLabelToIdMap: Map<string, string>;
+}): { pass: boolean; errorCode: selectionCriteriaErrorCode } {
+  const { groupby_dims, marker_score } = markerGene;
+
+  // When groupby_dims.tissue_ontology_term_label is undefined, default to ALL_TISSUES
+  const {
+    organism_ontology_term_label,
+    tissue_ontology_term_label = ALL_TISSUES,
+  } = groupby_dims;
+
+  // 1. Filter by organism
+  if (organism_ontology_term_label !== selectedOrganismLabel)
+    return {
+      pass: false,
+      errorCode: selectionCriteriaErrorCode.OrganismMismatch,
+    };
+
+  // 2. Filter by tissue
+  // There are marker genes tissues labeled as "All Tissues" AND there
+  // there are marker genes tissues that are undefined which are re-labeled
+  // as ALL_TISSUES. Select them only when selectedOrganLabel is "All Tissues"
+  let pass = false;
+
+  if (selectedOrganLabel === ALL_TISSUES) {
+    pass = tissue_ontology_term_label === ALL_TISSUES;
+  } else {
+    const tissueId = allTissuesLabelToIdMap.get(tissue_ontology_term_label);
+    pass = Boolean(
```

Comment:
nit:
You don't need to cast to Boolean since the && operator will already give you a true or a false

Reply:
The reason I did that is because in `line 61`, I initialize:

```
let pass = false;
```

That is a `boolean` type. `tissueId` could be of type `undefined`.

If you think it is better style, I could change the initialization of `pass` and remove the `Boolean` cast:

```
let pass: boolean | undefined = false
```

Would that be better?

Reply:
Sorry, I forgot to add that you could use `!!` to convert objects into boolean based on whether they are truthy or not.

Reply:
Anyway, just a nit, no changes needed really

Reply:
!! converts truthy or falsey to boolean.

Reply:
will change to `!!` since it is the preferred style